File: Python/Software Development Module - Python/Week5/ServiceApp.py
"""
ServiceApp.py
Billy Cussen
27/09/2020
"""

import gc
import Service
from Service import myService

#Variables
repeat = "y"

while repeat == "y":

    inputCheck = True

    while inputCheck:
        try:
            serviceType = str(input("Please enter the Type of Service You Would like (\"Pre\",\"Interim\" or \"Full\"): "))
            serviceType = serviceType.lower()
        except:
            print("You need to enter one of the Above listed Types")
        else:
            if serviceType != "pre" and serviceType != "interim" and serviceType != "full":
                print("You need to enter \"Pre\",\"Interim\" or \"Full\"")
            else:
                inputCheck = False

    while inputCheck == False:
        try:
            serviceLocation = str(input("Please enter where you want to Attend (\"Drogheda\" or \"Blanchardstown\"): "))
            serviceLocation = serviceLocation.lower()
        except:
            print("You need to enter one of the Above listed Types")
        else:
            if serviceLocation != "drogheda" and serviceLocation != "blanchardstown":
                print("You need to enter \"Drogheda\" or \"Blanchardstown\"")
            else:
                inputCheck = True

    myServiceApp = myService(serviceType,serviceLocation)
    print(myServiceApp.computeAndGetServiceDay())
    myServiceApp = None
    gc.collect()
    
    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter if you would like to repeat(\"Y\" or \"N\"): "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print("Please enter \"Y\" or \"N\"")
    
print ("Thanks for using my App!")

File: Java/Software Development Module - Java/Week5/ServiceApp.java
/*
ServiceApp.java
Billy Cussen
10/07/2020
*/

import javax.swing.JOptionPane;
import java.util.InputMismatchException;

public class ServiceApp {
    public static void main(String [] args){

        //Variables
        String branch = new String();
        String serviceType = new String();
        String day;
        boolean inputCheckService;

        //Objects
        Service myService;

        //Input
        do{
            inputCheckService = true;

            try{
                branch = JOptionPane.showInputDialog("Which Branch will you be visiting for your Service?\n 1. Blanchardstown, 2. Drogheda");
                
                if(!(branch.equalsIgnoreCase("Blanchardstown")||branch.equalsIgnoreCase("Drogheda"))){
                    throw new InputMismatchException("Please enter one of two stated branches!");
                }
            
            }catch(InputMismatchException e){
                JOptionPane.showMessageDialog(null, e.getMessage());
                inputCheckService = false;
            }

        }while(!inputCheckService);

        do{
            inputCheckService = true;

            try{
                serviceType = JOptionPane.showInputDialog("What Service Type would you like to visit?\n 1. Full, 2. Interim, 3. Pre-NCT");
                
                if(!(serviceType.equalsIgnoreCase("Full")||serviceType.equalsIgnoreCase("Interim")||serviceType.equalsIgnoreCase("Pre-NCT"))){
                    throw new InputMismatchException("Please enter one of three stated Service Types!");
                }
            
            }catch(InputMismatchException e){
                JOptionPane.showMessageDialog(null, e.getMessage());
                inputCheckService = false;
            }

        }while(!inputCheckService);

        //Process
        myService = new Service(branch, serviceType);
        myService.computeDay();

        //Output
        day = myService.getDay();
        myService = null;
        JOptionPane.showMessageDialog(null,"Your service Day is: "+day);

    }

}